src/solver/conv_hip_implicit_gemm_v4r4_gen_xdlops.cpp

```diff
@@ -328,10 +330,61 @@ static inline ConvSolution GetSolutionBase(const ConvolutionContext& ctx,
         }
     }
 
+    result.construction_params.push_back(construction_parameters);
+    const auto& dwDesc = ctx.conv_problem.GetWeights();
+
     if(ctx.direction.IsForward() || ctx.direction.IsBackwardData())
+    {
         result.invoker_factory = conv::MakeImplGemmDataInvokerFactory(ctx);
+    }
+    else if(dwDesc.GetType() == miopenHalf || dwDesc.GetType() == miopenBFloat16)
+    {
+        const auto lowp_quant  = ctx.conv_problem.GetConv().lowp_quant;
+        result.invoker_factory = [=](const std::vector<Kernel>& kernels) {
+            return [=](const Handle& handle, const boost::any& primitve_params) {
+                const auto invoke_params = boost::any_cast<conv::WrWInvokeParams>(primitve_params);
+                const auto& tensors      = invoke_params.tensors;
+                float zero               = 0.f;
+                TensorDescriptor workSpaceDesc(
+                    miopenFloat, tensors.dwDesc.GetLengths(), tensors.dwDesc.GetStrides());
+                SetTensor(handle, workSpaceDesc, invoke_params.workSpace, &zero);
+                float elapsed = std::numeric_limits<float>::max();
+                if(handle.IsProfilingEnabled())
+                    elapsed = handle.GetKernelTime();
+
+                handle.Run(kernels[0])(tensors.x, tensors.dy, invoke_params.workSpace);
+                if(handle.IsProfilingEnabled())
+                    elapsed += handle.GetKernelTime();
+
+                CastTensor(handle,
+                           &lowp_quant,
+                           workSpaceDesc,
+                           invoke_params.workSpace,
+                           tensors.dwDesc,
+                           tensors.dw,
+                           0,
+                           0);
+
+                if(handle.IsProfilingEnabled())
+                {
+                    elapsed += handle.GetKernelTime();
+                    handle.ResetKernelTime();
+                    handle.AccumKernelTime(elapsed);
+                }
+            };
+        };
+    }
+    else
+    {
+        result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+            return [=](const Handle& handle, const boost::any& primitve_params) {
```

Comment:
SetTensor is missing for fp32. Please take a look at the deleted line 4212 in convolutionocl.cpp file in this PR

Reply:
Thank you, will fix that.

Reply:
OMG

Reply:
I wonder how has it passed tests than...

Reply:
It didn't won in your tests. This issue leads to correctness issues.

Reply:
I mean there are long tests which should test something?

Reply:
It seems like this solver is a new iGemm that produces much more "universal" kernels (like Winograd, so that one kernel fits many problem configs) but the cost is performance drop. So it is quite possible that it never wins in our CI tests.

/cc @daniellowell

Reply:
Right, what would be interesting to have is a Find mode that also does verification. A self-testing feature.

Reply:
Interesting idea :fire:

Reply:
Actually we can make it so right now. The driver could register callbacks in the library. The library should invoke callback after each _Invok_ation. The driver verifies output buffer. Done. Ha.